src/OrbitGl/App.cpp

```diff
@@ -2991,7 +2999,19 @@ OrbitApp::GetConfidenceIntervalEstimator() const {
   return confidence_interval_estimator_;
 }
 
-void OrbitApp::ShowHistogram(const std::vector<uint64_t>* data, const std::string& function_name,
-                             uint64_t function_id) {
-  main_window_->ShowHistogram(data, function_name, function_id);
-}
\ No newline at end of file
+void OrbitApp::ShowHistogram(const std::vector<uint64_t>* data, const std::string& scope_name,
+                             uint64_t scope_id) {
+  main_window_->ShowHistogram(data, scope_name, scope_id);
+}
+
+[[nodiscard]] uint64_t OrbitApp::ProvideId(const orbit_client_protos::TimerInfo& timer_info) const {
+  if (timer_info.function_id() != orbit_grpc_protos::kInvalidFunctionId) {
```

Comment:
I am still thinking if it would be better to move this code into the IdProvider. So you have a uniform handling of scope ids.

Reply:
Discussed offline. Making it handle function ids too is too problematic, as it doesn't have access to function pretty names.